QuickOSM/ui/query_panel.py

```diff
@@ -118,13 +135,39 @@ def _run(self):
             bbox=properties['bbox'])
         self.end_query(num_layers)
 
-    def generate_query(self):
+    def generate_query_xml(self):
         query = self.dialog.text_query.toPlainText()
         area = self.dialog.places_edits[Panels.Query].text()
         self.write_nominatim_file(Panels.Query)
         properties = self.gather_values()
-        query = QueryPreparation(query, properties['bbox'], area)
+        server = get_setting('defaultOAPI', OVERPASS_SERVERS[0]) + 'convert'
+        query = QueryPreparation(query, properties['bbox'], area, server)
         query_string = query.prepare_query()
+        if query.is_oql_query():
+            url = query.prepare_url(QueryLanguage.XML)
+            connexion_overpass_api = ConnexionOAPI(url)
+            LOGGER.debug('Encoded URL: {}'.format(url))
+            query_string = connexion_overpass_api.run_convert()
+            query_string = html.unescape(query_string)
+
+        self.dialog.text_query.setPlainText(query_string)
+
+    def generate_query_oql(self):
+        query = self.dialog.text_query.toPlainText()
+        area = self.dialog.places_edits[Panels.Query].text()
+        self.write_nominatim_file(Panels.Query)
+        properties = self.gather_values()
+        server = get_setting('defaultOAPI', OVERPASS_SERVERS[0]) + 'convert'
+        query = QueryPreparation(query, properties['bbox'], area, server)
+        query_string = query.prepare_query()
+        if not query.is_oql_query():
+            query.prepare_query()
+            url = query.prepare_url(QueryLanguage.OQL)
+            connexion_overpass_api = ConnexionOAPI(url)
+            LOGGER.debug('Encoded URL: {}'.format(url))
+            query_string = connexion_overpass_api.run_convert()
+            query_string = html.unescape(query_string)
+
```

Comment:
I think you can compress these 2 functions into a single one. 80% of the lines are the same no?